### PR TITLE
Remove obsolete FunctionCode::Disconnect

### PR DESCRIPTION
**Pull request by uklotzde**
Use AsyncWriteExt::shutdown() to release all resources.

### CODE REVIEW

File: src/frame/mod.rs [hunk 1]
```diff
@@ -181,17 +174,6 @@ pub enum Request<'a> {
     /// The first parameter is the Modbus function code.
     /// The second parameter is the raw bytes of the request.
     Custom(u8, Cow<'a, [u8]>),
-
-    /// A poison pill for stopping the client service and to release
-    /// the underlying transport, e.g. for disconnecting from an
-    /// exclusively used serial port.
-    ///
-    /// This is an ugly workaround, because `tokio-proto` does not
```
Comment (uklotzde) [inline src/frame/mod.rs:189]: Does no longer apply.

**Comment by uklotzde** [comment]
Anyone? I only did a quick test with a USB serial adapter (RTU), basically runing the RTU example code in a loop.

Unfortunately, I don't remember the details what didn't work in the past with `tokio-proto`.

**Comment by uklotzde** [comment]
We should probably merge this PR and wait for reports from users that are already using `main`. Needs some testing before a release.

**Review by flosse — APPROVED** [review]

**Comment by uklotzde** [comment]
Let's give it a try.

**Pull request merged**